modules/iam/data.tf

```diff
@@ -2,7 +2,444 @@ data "aws_caller_identity" "current" {}
+    resources = ["arn:aws:iam::*:role/${var.security_audit_role_name}"]
+    actions   = ["sts:AssumeRole"]
+  }
+
+  statement {
+    sid       = "PutCloudWatchMetrics"
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["cloudwatch:PutMetricData"]
+  }
+
+  statement {
+    sid    = "SNS"
+    effect = "Allow"
+
+    resources = [
+      "arn:aws:sns:*:*:${var.project}-${var.environment}",
+      "arn:aws:sns:*:*:${var.project}-dlq-${var.environment}",
+    ]
+
+    actions = [
+      "sns:Publish",
+      "sns:Subscribe",
+    ]
+  }
+
+  statement {
+    sid       = "KMSforSNS"
+    effect    = "Allow"
+    resources = ["${var.kms_arn}"]
+
+    actions = [
+      "kms:Decrypt",
+      "kms:Encrypt",
+      "kms:GenerateDataKey",
+    ]
+  }
+
+  statement {
+    sid    = "DynamoDB"
+    effect = "Allow"
+
+    resources = [
+      "${local.ddb_table_arn}",
+      "${local.ddb_ip_table_arn}",
+    ]
+
+    actions = [
+      "dynamodb:DescribeTable",
+      "dynamodb:GetItem",
+      "dynamodb:PutItem",
+      "dynamodb:Query",
+      "dynamodb:Scan",
+      "dynamodb:UpdateItem",
+    ]
+  }
+}
+
+
+data "aws_iam_policy_document" "takeover" {
+  statement {
+    sid       = "WriteToCloudWatchLogs"
+    effect    = "Allow"
+    resources = ["arn:aws:logs:*:*:*"]
+
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+  }
+
+  statement {
+    sid       = "PutCloudWatchMetrics"
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["cloudwatch:PutMetricData"]
+  }
+
+  statement {
+    sid    = "SNS"
+    effect = "Allow"
+
+    resources = [
+      "arn:aws:sns:*:*:${var.project}-${var.environment}",
+      "arn:aws:sns:*:*:${var.project}-dlq-${var.environment}",
+    ]
+
+    actions = [
+      "sns:Publish",
+      "sns:Subscribe",
+    ]
+  }
+
+  statement {
+    sid       = "KMSforSNS"
+    effect    = "Allow"
+    resources = ["${var.kms_arn}"]
+
+    actions = [
+      "kms:Decrypt",
+      "kms:Encrypt",
+      "kms:GenerateDataKey",
+    ]
+  }
+
+  statement {
+    sid       = "GetAccountName"
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["iam:ListAccountAliases"]
+  }
+
+  statement {
+    sid    = "IAMPermissionsForElasticBEanstalk"
+    effect = "Allow"
+
+    resources = [
+      "arn:aws:iam::*:role/${var.project}*",
+      "arn:aws:iam::*:instance-profile/${var.project}*",
+    ]
+
+    actions = [
+      "iam:AddRoleToInstanceProfile",
+      "iam:AttachRolePolicy",
+      "iam:CreateInstanceProfile",
+      "iam:CreateRole",
+      "iam:DeleteInstanceProfile",
+      "iam:DeleteRole",
+      "iam:DetachRolePolicy",
+      "iam:GetInstanceProfile",
+      "iam:GetRole",
+      "iam:GetRolePolicy",
+      "iam:PassRole",
+      "iam:RemoveRoleFromInstanceProfile",
+      "iam:TagInstanceProfile",
+      "iam:TagRole",
+      "iam:UntagInstanceProfile",
+      "iam:UntagRole",
+      "iam:UpdateRole",
```

Comment:
Wow, this is a lot of permission. How come elasticbeanstalk needs all these?

cc: @paulschwarzenberger

Reply:
ElasticBeanstalk effectively uses EC2, creating instances with instance profiles and a load balancer, and it has to pass the IAM role to the EC2 instances